### ISSUE TITLE
os matches more files in Obsidian vault than oss

### ISSUE THREAD

**Issue by wenlzhang** [hauselin/obsidian-alfred#10]
Assume that we have the following notes:

- Test A.md (inside Obsidian vault)
- Test B.md (inside Obsidian vault)
- Test C.md (outside Obsidian vault)

The following tests are done with obsidian-alfred version 0.2.1.

If I search for the keyword "Test" with `os Test`, all three notes would be listed. However, if I search for the keyword "Test" with `oss Test`, only `Test A.md` and `Test C.md` would be listed.

Also, is it possible to open a note outside Obsidian vault, i.e. `Test C.md` in this case? Apparently one cannot open this note in Obsidian, but is it possible to open it with some default editor for Markdown files?

**Comment by hauselin** [comment]
> If I search for the keyword "Test" with os Test, all three notes would be listed. However, if I search for the keyword "Test" with oss Test, only Test A.md and Test C.md would be listed.

Yes, this is the expected behavior, as described in the README. I changed the behavior in response to an issue I closed earlier today (#9).

> Also, is it possible to open a note outside Obsidian vault, i.e. `Test C.md` in this case? Apparently one cannot open this note in Obsidian, but is it possible to open it with some default editor for Markdown files?

Yes, this should be relatively easy to implement. It should mostly involve replacing obsidian's uri (`obsidian://open?path=`) with `file:///` if your file is outside the vault. I'll work on it in the next few days.

**Comment by wenlzhang** [comment]
I don't fully understand the answer to the first question. Since both `Test A.md` and `Test B.md` are within Obsidian vaults. Shouldn't `os` and `oss` both list them when searching with the keyword "Test"?

After I switch on the option "with space" for `os`. Searching with `oss` tries to implement fallback search, as can be seen from the figure below. However, it would try to search for "oss Test" in my vault, instead of "Test". Also, I would have to hit Enter to select a specific vault to search the keyword in. Is it possible to search for the keyword in all vaults?
<img width="889" alt="Screenshot 2020-12-10 at 13 32 02" src="https://user-images.githubusercontent.com/44930227/101725750-7adf2980-3ab1-11eb-8d63-2e8333fc67e4.png">

**Comment by hauselin** [comment]
Oh, if you read the new documentation, it says you should press enter after typing `oss` (and the other two). That is, `oss`, press enter/return, then perform your search. So what you did with `oss Test` won't work because by default no arguments required, but you can make it work if you prefer (see next paragraph).

As stated in #9, there isn't a better solution than this for now (as far as I know). To restrict the search scope to your vaults, it requires an extra step. If you want to change its behavior slightly, set it to `Argument Optional`. In that case, it won't default to fallback search, but matching files also won't be listed until you press enter. There's no quick solution to this (unless the you/the user manually specifies the scopes in the File Filter `os`, `or`, `ot`). I can look into solutions with `mdfind`, but that would involve making significant changes to the current script/solution. 

My guess is many users should be using `os` and prefer matching results to be shown immediately (without having to press the extra enter/return) and also won't mind as much the fact that it lists files that are not inside the vault.

![fix](https://user-images.githubusercontent.com/7620977/101787053-a8859c00-3acc-11eb-85f0-40a78ccc410c.png)

**Comment by wenlzhang** [comment]
Ah, I see now. Thanks for the detailed explanation. I realize that I need to change the default keyword in the left part of the following blocks from `os` to `oss`. Now it works for me. I was looking for the keyword `oss` in the workflow blocks but didn't find any...

The default keyword was `os` for me probably because I updated the workflow to version 0.2.1 from a previous version. I also tried to make a clean installation of the workflow of version 0.2.2, then the default keyword on the left was `oss`.

<img width="730" alt="Screenshot 2020-12-10 at 23 25 32" src="https://user-images.githubusercontent.com/44930227/101792414-f7e7be80-3b04-11eb-888b-e20278c185fb.png">
